### PR TITLE
Correct the position when a parent element has "transform".

### LINKED ISSUE #37: transform breaks position: fixed
If a parent of your sticky elements has a [transform](https://www.w3.org/TR/css-transforms-1/#transform-rendering), e.g. `-webkit-transform: translateZ(0px)`, it will mess up the `position: fixed` that this library relies on. Might be worth noting in the docs, since this tripped me up.

### PR DESCRIPTION
**Pull request by jacobweber**
Fix #37. See if any of the scroll element's parents have a "transform" property.
If so, find the position of the closest one, and subtract it from the
fixed element's position.

### CODE REVIEW

File: src/sticky.js [hunk 1]
```diff
@@ -232,14 +232,26 @@ export default class Sticky extends Component {
...
+
     const newState = {
       fixed,
       height: wrapperRect.height,
       styles: fixed ? buildStickyStyle(mode, this.props, {
-        boundaryTop: mode === 'bottom' ? boundaryRect.top : 0,
-        boundaryBottom: mode === 'top' ? boundaryRect.bottom : 0,
-        top: mode === 'top' ? scrollRect.top : 0,
-        bottom: mode === 'bottom' ? scrollRect.bottom : 0,
+		boundaryTop: mode === 'bottom' ? boundaryRect.top : 0,
```
Comment (gm0t) [inline src/sticky.js]: could you please fix padding here (tabsize=2)?

File: src/sticky.js [hunk 2]
```diff
@@ -232,14 +232,26 @@ export default class Sticky extends Component {
     const scrollRect = getRect(scrollElement);
     const fixed = this.isFixed(holderRect, wrapperRect, boundaryRect, scrollRect);
 
+	function getClosestTransformedParent(el) {
```
Comment (gm0t) [inline src/sticky.js]: It's better to move this method to helpers, otherwise this function will be recreated every time

**Comment by gm0t** [comment]
Also, could you please make it optional? `getComputedStyle` is quite expensive and I don't think that a lot of people need it.

**Comment by gm0t** [comment]
Oh, and of course thank you very much for your PR! I really appreciate your help!

**Comment by jacobweber** [comment]
No problem. I added a offsetTransforms property that defaults to false. It might be possible to speed this up by only calculating the offsets when you render, but this is working for now.

**Pull request merged**